src/main/java/digital/slovensko/autogram/server/dto/SignRequestBody.java

```diff
@@ -35,26 +55,28 @@ public InMemoryDocument getDocument() throws RequestValidationException {
         if (document.getContent() == null)
             throw new RequestValidationException("Document.Content is required", "");
 
-        byte[] content;
-        if (isBase64()) {
-            content = Base64.getDecoder().decode(document.getContent());
-        } else {
-            content = document.getContent().getBytes();
-        }
-
+        var content = decodeDocumentContent();
         var filename = document.getFilename();
-        var mimetype = AutogramMimeType.fromMimeTypeString(payloadMimeType.split(";")[0]);
 
-        return new InMemoryDocument(content, filename, mimetype);
+        return new InMemoryDocument(content, filename, getMimetype());
     }
 
-    public SigningParameters getParameters() throws RequestValidationException {
+    public SigningParameters getParameters() throws RequestValidationException, MalformedBodyException {
         if (parameters == null)
             throw new RequestValidationException("Parameters are required", "");
 
         parameters.validate(getDocument().getMimeType());
 
-        return parameters.getSigningParameters(isBase64());
+        var signingParameters = parameters.getSigningParameters(isBase64());
+        if (isAsice(getMimetype()) || isXML(getMimetype()) || isXDC(getMimetype()))
```

Comment:
Tu 3x parsujeme ten mime type, ulozme si to niekam.

Reply:
uložil som